### ISSUE TITLE
Allow SSL connection without providing certificates (for Heroku Postgres connection)

### ISSUE THREAD

**Issue by sdil** [bitnami/bitnami-docker-pgbouncer#26]
**Description**

In order to connect from pgbouncer to Heroku Postgres instance, I'll need to use SSL connection. However, I do not have the cert (to be honest, I'm not really sure how TLS cert works)

**Steps to reproduce the issue:**

1. Launch docker container with the following command:

```
docker run \
    -p 6432:6432 \
    --name=pgbouncer \
    -e "POSTGRESQL_HOST=<Heroku Postgres URL>" \
    -e "POSTGRESQL_PORT=5432" \
    -e "POSTGRESQL_USERNAME=xxx" \
    -e "POSTGRESQL_PASSWORD=xxx" \
    -e "POSTGRESQL_DATABASE=xxx" \
    -e "PGBOUNCER_POOL_MODE=transaction" \
    -e "PGBOUNCER_SERVER_TLS_SSLMODE=require" \
    -e "PGBOUNCER_SERVER_TLS_PROTOCOLS=secure" \
    bitnami/pgbouncer
```

**Describe the results you received:**

This will not work and give the following error message:

```
pgbouncer 15:56:21.17 
pgbouncer 15:56:21.18 Welcome to the Bitnami pgbouncer container
pgbouncer 15:56:21.18 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-pgbouncer
pgbouncer 15:56:21.18 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-pgbouncer/issues
pgbouncer 15:56:21.18 
pgbouncer 15:56:21.21 INFO  ==> ** Starting PgBouncer setup **
pgbouncer 15:56:21.23 INFO  ==> Validating settings in PGBOUNCER_* env vars...
pgbouncer 15:56:21.23 ERROR ==> You must provide a X.509 certificate in order to use server TLS
pgbouncer 15:56:21.24 ERROR ==> You must provide a private key in order to use server TLS
pgbouncer 15:56:21.24 WARN  ==> A CA X.509 certificate was not provided. Server verification will not be performed in TLS connections
```
**Describe the results you expected:**

It should allow users to connect to Heroku Postgre instance with SSL mode without the SSL cert. I used `pgbouncer/pgbouncer` image and it works fine like below:

```
sdil@sdil-ThinkPad-T470p ~ $ docker run \
>     -p 6431:6432 \
>     --name=pgbouncer \
>     -e "DATABASES_HOST=xxx" \
>     -e "DATABASES_PORT=5432" \
>     -e "DATABASES_USER=xxx" \
>     -e "DATABASES_PASSWORD=xxx" \
>     -e "DATABASES_DBNAME=xxx" \
>     -e "PGBOUNCER_LISTEN_PORT=6432" \
>     -e "PGBOUNCER_POOL_MODE=transaction" \
>     -e "PGBOUNCER_SERVER_TLS_SSLMODE=require" \
>     -e "PGBOUNCER_SERVER_TLS_PROTOCOLS=secure" \
>     pgbouncer/pgbouncer
Creating pgbouncer config in /etc/pgbouncer
#pgbouncer.ini

[databases]
* = host = xxx port=5432 user=xxx password=xxx dbname=xxx

[pgbouncer]
listen_addr = 0.0.0.0
listen_port = 6432
auth_type = any
pool_mode = transaction
ignore_startup_parameters = extra_float_digits

# Log settings
admin_users = postgres

# Connection sanity checks, timeouts

# TLS settings
server_tls_sslmode = require
server_tls_protocols = secure

# Dangerous timeouts

Starting pgbouncer.
2022-01-06 15:57:28.087 UTC [1] LOG kernel file descriptor limit: 1048576 (hard: 1048576); max_client_conn: 100, max expected fd use: 112
2022-01-06 15:57:28.087 UTC [1] LOG listening on 0.0.0.0:6432
2022-01-06 15:57:28.087 UTC [1] LOG listening on unix:/tmp/.s.PGSQL.6432
2022-01-06 15:57:28.087 UTC [1] LOG process up: PgBouncer 1.15.0, libevent 2.1.8-stable (epoll), adns: c-ares 1.15.0, tls: LibreSSL 2.7.5
```

**Comment by carrodher** [comment]
It seems it is not an issue related to the Bitnami pgbouncer container image but about how the application or environment is being used/configured. [Here](https://github.com/bitnami/bitnami-docker-pgbouncer#securing-pgbouncer-traffic) you can find more info about parameters used to secure traffic, just in case the descriptions can help with your configuration.

For information regarding the application itself, customization of the content within the application, or questions about the use of technology or infrastructure; we highly recommend checking forums and user guides made available by the project behind the application or the technology.

That said, I will keep this ticket open until the stale bot closes it just in case someone from the community adds some valuable info.

**Comment by sdil** [comment]
[From line 98 in script libpgbouncer.sh](https://github.com/bitnami/bitnami-docker-pgbouncer/blob/master/1/debian-10/rootfs/opt/bitnami/scripts/libpgbouncer.sh#L98), it states that I can't launch the bitnami pgbouncer when setting `sslmode = require` unless I set the cert. If we were to allow user to set `sslmode = require` without specifying the cert, then I would be able to connect to the Heroku Postgres instance with SSL enabled.

Again, I'm not really familiar with how the TLS connection works in this case. If this request is not a valid one, you can ignore this issue and I'll use another container image on my end :pray:

**Comment by pawel-szon** [comment]
I'm experiencing the same issue. `PGBOUNCER_SERVER_TLS_SSLMODE = require` should not require CA, cert and key files since effectively it's not validating server's certificate.

**Comment by fmulero** [comment]
Hi @pawel-szon and @sdil , we recently released a new version (see bitnami/bitnami-docker-pgbouncer#27) to disable the validation of `PGBOUNCER_SERVER_TLS_CERT_FILE` and `PGBOUNCER_SERVER_TLS_KEY_FILE` when `PGBOUNCER_SERVER_TLS_SSLMODE = verify-ca`. In that case you have to set only `PGBOUNCER_SERVER_TLS_CA_FILE` I hope it helps. In the meantime we are going to open an internal discussion to review all this configuration, but feel free to open PR with your changes.
Regards

**Comment by fmulero** [comment]
Hi @pawel-szon and @sdil, new version was recently released to amend this issue, `PGBOUNCER_SERVER_TLS_CERT_FILE` and `PGBOUNCER_SERVER_TLS_KEY_FILE` validations where changed accoding to pgbouncer documentation

**Comment by sdil** [comment]
@fmulero thank you!

**Comment by spenceclark** [comment]
Has this issue regressed in the last few days? I'm using the :latest tag on the docker image and my pgbouncer pods all started failing with the validation error message a few hours ago. I've fixed the tag to "1.16.1-debian-10-r71" from a week ago, and its working ok again now.

**Comment by fmulero** [comment]
Yes @spenceclark , you right. We are working on it
